### ISSUE TITLE
CMS Preview Draft 500 error when using cacheinclude's main.php

### ISSUE THREAD

**Issue by christopherdarling** [heyday/silverstripe-cacheinclude#23]
When using the default framework/main.php everything works fine, when using the cacheinclude/main.php I get a 500 error when trying to preview in draft mode.

I've tracked this down to DB::get_conn() returning null which is called by ExactMatchFilter from Member::currentUser()

Stack trace below;

```
Fatal error: Call to a member function comparisonClause() on null in G:\sites\www\framework\search\filters\ExactMatchFilter.php on line 34
Call Stack
#   Time    Memory  Function    Location
1   0.0010  143528  {main}( )   ...\main.php:0
2   0.1355  5195416 RequestProcessor->preRequest( ) ...\main.php:24
3   0.1355  5195648 VersionedRequestFilter->preRequest( )   ...\RequestProcessor.php:33
4   0.1405  5492136 Versioned::can_choose_site_stage( ) ...\VersionedRequestFilter.php:19
5   0.1549  6445784 Member::currentUser( )  ...\Versioned.php:1107
6   0.2465  8469168 DataList->byID( )   ...\Member.php:784
7   0.2465  8469248 DataList->filter( ) ...\DataList.php:872
8   0.2465  8469664 DataList->addFilter( )  ...\DataList.php:370
9   0.2466  8470056 DataList->applyFilterContext( ) ...\DataList.php:386
10  0.2486  8556312 DataList->alterDataQuery( ) ...\DataList.php:534
11  0.2487  8556752 call_user_func:{G:\sites\www\framework\model\DataList.php:131} ( )  ...\DataList.php:131
12  0.2487  8556832 SearchFilter->apply( )  ...\DataList.php:131
13  0.2487  8556944 ExactMatchFilter->applyOne( )   ...\SearchFilter.php:217
```

**Comment by stecman** [comment]
What SilverStripe framework/cms version are you using, @christopherdarling?

**Comment by christopherdarling** [comment]
@stecman we are on 3.3.1

**Comment by bendubuisson** [comment]
Fixed by #25
